lib/src/core/gestures/drag.dart

```diff
@@ -1,655 +1,1055 @@
+      final double averageY = _resolveDeltaForPanGesture(axis: _DragDirection.vertical, localDelta: localDelta);
+      final Offset updatedDelta = Offset(averageX, averageY) - _lastUpdatedDeltaForPan;
+      _lastUpdatedDeltaForPan = Offset(averageX, averageY);
+      dx = updatedDelta.dx;
+      dy = updatedDelta.dy;
+    }
+
+    return Offset(dx, dy);
+  }
+
+  double _resolveDelta({
+    required int pointer,
+    required _DragDirection axis,
+    required Offset localDelta,
+  }) {
+    final bool positive = axis == _DragDirection.horizontal ? localDelta.dx > 0 : localDelta.dy > 0;
+    final double delta = axis == _DragDirection.horizontal ? localDelta.dx : localDelta.dy;
+    final int? maxSumDeltaPointer = _getMaxSumDeltaPointer(positive: positive, axis: axis);
+    assert(maxSumDeltaPointer != null);
+
+    if (maxSumDeltaPointer == pointer) {
+      return delta;
+    } else {
+      final double maxSumDelta = _getSumDelta(pointer: maxSumDeltaPointer!, positive: positive, axis: axis);
+      final double curPointerSumDelta = _getSumDelta(pointer: pointer, positive: positive, axis: axis);
+      if (positive) {
+        if (curPointerSumDelta + delta > maxSumDelta) {
+          return curPointerSumDelta + delta - maxSumDelta;
+        } else {
+          return 0.0;
+        }
+      } else {
+        if (curPointerSumDelta + delta < maxSumDelta) {
+          return curPointerSumDelta + delta - maxSumDelta;
+        } else {
+          return 0.0;
+        }
+      }
+    }
+  }
+
+  double _resolveDeltaForPanGesture({
+    required _DragDirection axis,
+    required Offset localDelta,
+  }) {
+    final double delta = axis == _DragDirection.horizontal ? localDelta.dx : localDelta.dy;
+    final int pointerCount = _acceptedActivePointers.length;
+    assert(pointerCount >= 1);
+
+    double sum = delta;
+    for (final Offset offset in _moveDeltaBeforeFrame.values) {
+      if (axis == _DragDirection.horizontal) {
+        sum += offset.dx;
+      } else {
+        sum += offset.dy;
+      }
+    }
+    return sum / pointerCount;
+  }
+
+  @override
+  void handleEvent(PointerEvent event) {
+    assert(_state != _DragState.ready);
+    if (!event.synthesized &&
+        (event is PointerDownEvent ||
+            event is PointerMoveEvent ||
+            event is PointerPanZoomStartEvent ||
+            event is PointerPanZoomUpdateEvent)) {
+      final Offset position = switch (event) {
+        PointerPanZoomStartEvent() => Offset.zero,
+        PointerPanZoomUpdateEvent() => event.pan,
+        _ => event.localPosition,
+      };
+      _velocityTrackers[event.pointer]!.addPosition(event.timeStamp, position);
+    }
+    if (event is PointerMoveEvent && shouldGiveUp != null && shouldGiveUp!(event)) {
+      _giveUpPointer(event.pointer);
+      return;
+    }
+    if (event is PointerMoveEvent && event.buttons != _initialButtons) {
+      resolve(GestureDisposition.accepted);
```

Comment:
I copied this code from what we had before, but it looks wrong to me.

The [flutter implementation](https://github.com/flutter/flutter/blob/1c9c375e4e33e638aa6be3a25e466f494cc1987e/packages/flutter/lib/src/gestures/monodrag.dart#L617-L618) rejects the gesture at this point, but we accept it.

```suggestion
    if (event is PointerMoveEvent && event.buttons != _initialButtons) {
      _giveUpPointer(event.pointer);
      return;
```

Reply:
shouldGiveUp is an intended hook into gesture recognizer that allows to give up specific gesture by a certain condition

In Sweyer this was needed in order to allow a whole-screen horizontal drawer opening gesture, because it conflicts with PageView horizontal gesture recognizer

When we drag left, when we drag right+we are at the begginning of TabBarView, we want it to give up gesture up to drawer, otherwise TabBarView should have the priority (which it has by default if it doesn't give up)

Is my understanding correct, that you updated this file with new changes from framework? GitHub diff view has gone crazy on this one...

Reply:
> shouldGiveUp is an intended hook into gesture recognizer that allows to give up specific gesture by a certain condition

`shouldGiveUp` allows us to reject an event early, and we do that in the if directly above this one. This if on the other hand handles when the user switches the mouse button during the move. The [Flutter implementation has the exact same check](https://github.com/flutter/flutter/blob/1c9c375e4e33e638aa6be3a25e466f494cc1987e/packages/flutter/lib/src/gestures/monodrag.dart#L617-L618), but they abort the gesture detection at this point, which makes sense I think. Our version instantly accept the gesture here when we switch the button, which seems like a mistake.

> Is my understanding correct, that you updated this file with new changes from framework? GitHub diff view has gone crazy on this one...

Yeah, I updated it to commit `d2b42d8c13f2dba6ba12c84a0e66fadf7eb94c16`. The diff is indeed crazy, but it get's much better to review if you [turn of whitespace changes](https://github.com/nt4f04uNd/nt4f04unds_widgets/pull/12/commits/0d78622890a787e7b03ad08d2e63057235f82631?diff=unified&w=1).

Reply:
Ah, I see. Honestly, I don't remember why it's there. My only guess right now is that I made it to reduces the slugginess of gesture fighting.

Right now it doesn't seem like there's any "dead area" for drawer gesture ([link](https://github.com/nt4f04uNd/sweyer/blob/adjust-workflow-runs/lib/widgets/drawer.dart#L74)), so I decided to not build anything fancy on top and just resolve it to accepted.

Reply:
Well I don't think there si a problem in having the `shouldGiveUp`, but don't you think we should reject a gesture if the user switches the mouse button (I know this is not really a problem right now because we only have mobile, but you can use a real mouse on Android). Shouldn't we fix this `if` like so?:

```suggestion
    if (event is PointerMoveEvent && event.buttons != _initialButtons) {
      _giveUpPointer(event.pointer);
      return;
```

Reply:
Yeah, let's fix it